### ISSUE TITLE
Link formatting broken in v0.6.0?

### ISSUE THREAD

**Issue by vyder** [piotrmurach/tty-markdown#26]
### Describe the problem

I am planning to use `tty-markdown` in a standalone ruby script - and I have found that links are not formatted correctly in the version that's on RubyGems (`v0.6.0`).

I have also tried making a build of `v0.7.0` of this repo from `master` at `d5a790e` - and that actually throws an error about `URI` symbol not found - and adding a `require 'uri'` to the top of `lib/tty/markdown/converter.rb` fixes the issue and formats the links correctly from the test script.

I believe your `link_spec.rb` is not failing on this issue because likely some other gem in the test suite requires `URI`. I also couldn't think of a way to write a test case for this issue.

I'm creating a PR for this - it's a one-line change really - let me know if you prefer solving this another way, will be happy to code that up and write a test.

### Steps to reproduce the problem

```ruby
markdown-test
---
#!/usr/bin/env ruby

require 'tty-markdown'

markdown =<<-TEXT

## Inline link

[TTY Toolkit](https://ttytoolkit.org)

## Inline link with title

[TTY site](https://ttytoolkit.org "TTY Toolkit")

## Inline text matching link

[https://ttytoolkit.org](https://ttytoolkit.org)

## Inline link reference

[TTY Toolkit][1]

## Email link

[Email me](mailto:test@example.com)

[1]: https://ttytoolkit.org

TEXT

print TTY::Markdown.parse(markdown)
```

### Actual behaviour

My output (`v0.6.0` from RubyGems):

![Screen Shot 2020-08-26 at 12 29 44 pm](https://user-images.githubusercontent.com/587136/91280080-ed96f400-e7a3-11ea-8bd5-d8c6bc6498f7.png)

`v0.7.0` from `master` at `d5a790e`:

![Screen Shot 2020-08-26 at 2 01 40 pm](https://user-images.githubusercontent.com/587136/91280911-153a8c00-e7a5-11ea-91a7-5fc06d7f1df8.png)

With `require 'uri'` in `converter.rb`:

![Screen Shot 2020-08-26 at 2 03 02 pm](https://user-images.githubusercontent.com/587136/91280932-1bc90380-e7a5-11ea-8617-f7ae54806b40.png)

The error is specifically:

```
[...]/gems/tty-markdown-0.7.0/lib/tty/markdown/converter.rb:649:in `convert_a': uninitialized constant TTY::Markdown::Converter::URI (NameError)
```

### Expected behaviour
Expected the links to be formatted per the test spec - both label/title and url is visible.

### Describe your environment

* OS version: macOS 10.13.6
* Ruby version: 2.5.0
* TTY::Markdown version: 0.6.0 / 0.7.0

**Comment by piotrmurach** [comment]
Hi Vidur,

Thanks for this report and using `tty-markdown`.

I'm continuously working on making the next release soon which is going to be the biggest release yet. I set out to fix few issues but ended up rewriting all the internals(see [changelog](https://github.com/piotrmurach/tty-markdown/blob/master/CHANGELOG.md#v070---unreleased)). I would say I'm very close but still need to make a few major changes on how conversion works with maintenance and performance in mind. 

Not sure if you're aware but there is a companion project [tty-markdown-cli](https://github.com/piotrmurach/tty-markdown-cli) that will be released when this gem is published. The reason I mention it is that you said you're writing a script. It may be that what you're trying to do fits well with the `tty-markdown-cli` scope. Please have a look.

**Comment by vyder** [comment]
You're most welcome Piotr!

Ah I see - very cool. Yes when I first cloned the repo I was unaware that I was looking at v0.7.0 and was confused about the difference in functionality 🙈 - but I gotta say it's organized very cleanly, it took me literally just a few mins to figure out what module was doing what.

I am aware of the CLI gem, but this library solves my use case perfectly. Do you have a timeline for when v0.7.0 will be out? Just curious, no rush, I will use the git repo as the gem source till then.

Thanks for your work! I also love the simplicity of your `finite_machine` gem, I recently replaced my hacky FSM with yours and was pleasantly surprised to find that tty toolkit is also written by you.

**Comment by piotrmurach** [comment]
As for the release date, I cannot pin down anything, but all I can say this is my sole focus in open source now until the release. I'd say I'm 80% done. It all depends on time and if I uncover any other issues. This project keeps on giving ;-) I needed to fix one issue in the table rendering and few more appeared which I couldn't ignore. I'm not aiming for perfection but equally, I don't want to release something I know is broken. I also want to be happy with how it handles README files which it now can parse pretty nicely. I still want to add footnotes support as well.

Glad to hear that `tty-markdown` matches your needs. Using the master branch is probably best at the moment as you will be 'prepared' when I make a release. and I'm also open to feedback.

**Comment by piotrmurach** [comment]
Released `v0.7.0` with many updates(see [changelog](https://github.com/piotrmurach/tty-markdown/blob/master/CHANGELOG.md)). I hope you enjoy it!
